From 99529e584b3605c040ad5ad88792a28a2c1ff263 Mon Sep 17 00:00:00 2001
From: Marco Martin <notmart@gmail.com>
Date: Wed, 15 Jun 2022 11:13:04 +0200
Subject: [PATCH] fix windows dragging in desktop grid

Don't disable the main drag handler when canclosewindows is disabled,
that one is not used to close windows but to drag them on other desktops or screens

BUG: 455268


(cherry picked from commit e04542995e45d29c4b4ef8664a6653afa112cfd6)
---
 src/effects/private/qml/WindowHeap.qml | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/effects/private/qml/WindowHeap.qml b/src/effects/private/qml/WindowHeap.qml
index 230f86fc7..66b45e799 100644
--- a/src/effects/private/qml/WindowHeap.qml
+++ b/src/effects/private/qml/WindowHeap.qml
@@ -396,8 +396,8 @@ FocusScope {
 
                 component DragManager : DragHandler {
                     id: dragHandler
-                    enabled: heap.supportsCloseWindows
                     target: null
+                    grabPermissions: PointerHandler.CanTakeOverFromHandlersOfSameType
 
                     readonly property double targetScale: {
                         if (!heap.supportsDragUpGesture) {
@@ -432,6 +432,7 @@ FocusScope {
                 DragManager {
                     id: touchDragHandler
                     acceptedDevices: PointerDevice.TouchScreen
+                    enabled: heap.supportsCloseWindows
                     readonly property double targetOpacity: {
                         if (!heap.supportsCloseWindows) {
                             return 1;
-- 
2.36.1

